faux-mgs/src/main.rs

```diff
@@ -1457,6 +1502,79 @@ async fn monorail_unlock(
         UnlockChallenge::Trivial { timestamp } => {
             UnlockResponse::Trivial { timestamp }
         }
+        UnlockChallenge::EcdsaSha2Nistp256(data) => {
+            let keys = ssh_list_keys()?;
+            let pub_key = if keys.len() == 1 {
+                keys[0].clone()
+            } else {
+                let Some(pub_key) = pub_key else {
+                    bail!(
+                        "need --key for ECDSA challenge; \
+                         multiple keys are available"
+                    );
+                };
+                if pub_key.ends_with(".pub") {
+                    ssh_key::PublicKey::read_openssh_file(Path::new(&pub_key))
+                        .with_context(|| {
+                        format!("could not read key from {pub_key:?}")
+                    })?
+                } else {
+                    let mut found = None;
+                    for k in keys.iter() {
+                        if k.to_openssh()?.contains(&pub_key) {
+                            if found.is_some() {
+                                bail!("multiple keys contain '{pub_key}'");
+                            }
+                            found = Some(k);
+                        }
+                    }
+                    let Some(found) = found else {
+                        bail!(
+                            "could not match '{pub_key}'; \
+                        use `faux-mgs monorail unlock --list` to print keys"
```

Comment:
nit - rustfmt gave up on indenting this multiline string

Reply:
fixed